test_scripts/WebEngine/028_PTU_snapshot_validation_no_app_connected.lua

```diff
@@ -0,0 +1,71 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0240-sdl-js-pwa.md
+--
+-- Description:
+-- Verify that PTU is performed after BC.SetAppProperties request with new application properties of the policyAppID
+-- Precondition:
+-- 1. SDL and HMI are started
+--
+-- Sequence:
+-- 1. HMI sends BC.SetAppProperties request with new application properties of the policyAppID to SDL
+--  a. SDL sends successful response to HMI
+--  b. PTU is triggered, SDL sends UPDATE_NEDDED to HMI
+--  с. PTS is created with application properties of the policyAppID and other mandatory pts
+--------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/WebEngine/commonWebEngine')
+local hmi_ptu = require('test_scripts/Policies/HMI_PTU/common_hmi_ptu')
+
+
+--[[ Local Variables ]]
+local  appStoreConfig = {
+  keep_context = false,
+  steal_focus = false,
+  priority = "NONE",
+  default_hmi = "NONE",
+  groups = { "Base-4" }
+}
+
+local appProperties = {
+  nicknames = { "Test Web Application_21", "Test Web Application_22" },
+  policyAppID = "0000002",
+  enabled = true,
+  authToken = "ABCD12345",
+  transportType = "WS",
+  hybridAppPreference = "CLOUD"
+}
+
+local appPropExpected = {
+  nicknames = { "Test Web Application_21", "Test Web Application_22" },
+  auth_token = "ABCD12345",
+  cloud_transport_type = "WS",
+  enabled = "true",
+  hybrid_app_preference = "CLOUD"
+}
+
+--[[ Local Functions ]]
+local function PTUfunc(tbl)
+  tbl.policy_table.app_policies[common.getConfigAppParams().fullAppID] = appStoreConfig;
+end
+
+local function setAppProperties(pData)
+  local corId = common.getHMIConnection():SendRequest("BasicCommunication.SetAppProperties",
+    { properties = pData })
+  common.getHMIConnection():ExpectResponse(corId,
+    { result = { code = 0 }})
+  common.isPTUStarted()
+  common.wait(1000)
```

Comment:
do we need to wait after calling isPTUstarted? is this really to keep verifyPTSnapshot from running too early?

Reply:
This was in the 026 test that this test was based off of. Yes I imagine it is there for stability purposes to allow core enough time to write the PTS to disk.

Reply:
ok, I was thinking if it has to do with verifying the PTS, maybe the wait should go at the beginning of that function but it's fine in this local function too 👍